src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -448,6 +451,26 @@ std::optional<Int32> IcebergMetadata::getSchemaVersionByFileIfOutdated(String da
     return std::optional{schema_id};
 }
 
+void IcebergMetadata::createInitial(
+    const ObjectStoragePtr & object_storage,
+    const StorageObjectStorageConfigurationWeakPtr & configuration,
+    const ContextPtr & local_context,
+    const std::optional<ColumnsDescription> & columns,
+    ASTPtr partition_by)
+{
+    auto configuration_ptr = configuration.lock();
+
+    const auto metadata_files = listFiles(*object_storage, *configuration_ptr, "metadata", ".metadata.json");
+    if (!metadata_files.empty())
+        return;
+
+    auto metadata_content = createEmptyMetadataFile(configuration_ptr->getPath(), *columns, partition_by, configuration_ptr->getDataLakeSettings()[DataLakeStorageSetting::iceberg_format_version]);
+    auto filename = configuration_ptr->getPath() + "metadata/v1.metadata.json";
+    auto buffer_metadata = object_storage->writeObject(
+        StoredObject(filename), WriteMode::Rewrite, std::nullopt, DBMS_DEFAULT_BUFFER_SIZE, local_context->getWriteSettings());
+    buffer_metadata->write(metadata_content.data(), metadata_content.size());
```

Comment:
BTW, where we specify that we want to do "create if not exists" (google for `If-None-Match` for S3, there similar requests for all other object storages)